File: 2277-count-equal-and-divisible-pairs-in-an-array/count-equal-and-divisible-pairs-in-an-array.cpp
#pragma GCC optimize("O2")

#define fast_io                   \
    ios_base::sync_with_stdio(0); \
    cin.tie(0);                   \
    cout.tie(0);

class Solution 
{
public:
    int countPairs(vector<int>& nums, int k) 
    {   
        fast_io;

        int count = 0;
        for (int i = 0; i < nums.size() - 1; ++i)
            for (int j = i + 1; j < nums.size(); ++j)
                if (nums[i] == nums[j] && (i * j) % k == 0)
                    ++count;

        return count;
    }
};

File: 2277-count-equal-and-divisible-pairs-in-an-array/count-equal-and-divisible-pairs-in-an-array.py
class Solution:
    def countPairs(self, nums: list[int], k: int) -> int:
        return sum(
            1 if nums[i] == nums[j] and (i * j) % k == 0 else 0
            for i in range(len(nums) - 1)
            for j in range(i + 1, len(nums))
        ) 